### ISSUE TITLE
[BUG] Execute httpEndPoint using WorkFlowInstanceID

### ISSUE THREAD

**Issue by Prameelagr8** [elsa-workflows/elsa-core#5879]
## Description
 Execute httpEndPoint using WorkFlowInstanceID

## Steps to Reproduce
To help  identify the issue more quickly, please follow these guidelines:

1. I have a workflow which has 2 httpEndPoints. One at start which is executed successfully using Postman and second one will wait until it is executed to continue rest of the actions. I have build a sample usecase for reference.

![image](https://github.com/user-attachments/assets/db3dc569-d87f-469b-a205-b6085d141db1)

![image](https://github.com/user-attachments/assets/50820725-d032-462a-aa50-6aad47538ea4)

2.Now, I have hit 1st httpEndpoint 3 times which made all 3 suspended activities. How can i get hit second httpEndPoint selectively using postman based on WorkFlowInstanceID? (For Example, I need to finish only 2nd one among all 3 suspended workflows. How to hit with WorkflowInstanceID?)

![image](https://github.com/user-attachments/assets/a96d4da2-46ef-4f2e-b52e-1377a30ae392)

****I need to hit this using postman. We are evaluating Elsa so as of now we are checking using postman all http concurrent activities.**

**Comment by Prameelagr8** [comment]
HttpEndPoint with Workflowid. Can someone check this issue and respond please

**Comment by Prameelagr8** [comment]
@sfmskywalker Please let me know how to trigger httpEndPoint using workflowInstanceID (To handle multiple instances suspended mode and resume by instanceID)

**Comment by sfmskywalker** [comment]
Hi @Prameelagr8 , thanks for the question! While I was looking for the answer, I encountered a bug that prevents you from doing this at the moment. As a temporary workaround, you can do it as follows:

- In your subsequent HTTP Endpoint activities, embed the workflow instance in the path using a JS expression.
- E.g.: `${getWorkflowInstanceId()}/next`
- The path would be e.g. `https://localhost:5001/workflows/971a574cf668a047/next`

This way, you will be able to resume the exact workflow instance that you need by sending an HTTP request to `https://localhost:5001/workflows/971a574cf668a047/next`.

As I mentioned, this is a temporary workaround that you can try. In the meantime, I am working on a fix. When fixed, you will be able to send the workflow instance as an HTTP header or as a query string parameter.

**Comment by Prameelagr8** [comment]
@sfmskywalker thank you for confirmation. Yes it used getWorkflowInstanceId() to form dynamic endpoint but approach is little different which is working fine in elsa 3 but not on 3.3.. I will post json export soon with my remaining questions.

We are exploring elsa to form complex workflows which is working fine as of now with elsa3 but its good if they work in 3.3 also which will allow us to use all functionalities.

**Comment by sfmskywalker** [comment]
Indeed I noticed some additional issues with HTTP workflows in 3.3-preview, which I'll investigate next. In the meantime #5908 fixes the issue so that you can provide the workflow instance ID as an header or a querystring parameter, instead of having to do so via the endpoint path.

Example using a header:

```
curl --location 'https://localhost:5001/workflows/next' \
--header 'X-Workflow-Instance-ID: ce78780ff8cf3d03' 
```

Using querystring:

```
curl --location 'https://localhost:5001/workflows/next?workflowInstanceId=ce78780ff8cf3d03'
```

I will proceed to close this issue as fixed, but please feel free to continue the conversation. For other issues and questions, please open separate issues.

Thanks again for reporting this issue.

**Comment by sfmskywalker** [comment]
Reopening until after the PR is reviewed & merged.

**Comment by sfmskywalker** [comment]
Fixed via #5908
